### DISCUSSION TITLE
FScope Board

### DISCUSSION THREAD

**Discussion by Jason-BCIT** [fhdm-dev/scoppy#68] [Scoppy Front end designs]
Thanks for everything you've built and shared with Scoppy. It's very cool.

I teach introductory electronics to future high school Technology Education ("shop") teachers in Vancouver, BC. In the past I have had them build the DPScope SE (http://dpscope.freevar.com/overview_se.html) as it is a great way to get a basic scope, but also to learn about some of the components that go into the scope. The author of that project has a great description of his front end, design choices, and an amazing assembly guide on his website.

I was in the midst of translating some of his front end design to work with Scoppy when I went on holidays... when I came back and checked for updates I noticed the fscope board mentioned in the wiki, but not information on how to acquire one or study the design in more detail.

I realize that a project like this can suck up a LOT of time, so my apologies if it is in the "more information coming soon" category. I'll carry on with my design as I'm aiming for a board with through-hole components and look forward to what you can share about the fscope board, when you've got time to share it.

Thanks,

Jason

P.S. After years of lurking on Github, the Scoppy project finally got me to sign up for an account and make a post... and I'll definitely be sharing my design once it is tested and working.

**Comment by fhdm-dev** [comment]
Hi Jason,

I assume you came across this: [https://oscilloscope.fhdm.xyz/wiki/fscope-250k5-v2](https://oscilloscope.fhdm.xyz/wiki/fscope-250k5-v2)
I didn't intend for that page to be public yet 'cos I haven't quite finished setting up the online store and I currently don't have any of the boards in stock. Oops!
     
As you can see from the photo the board is mainly SMD components so the only soldering required is some pin headers, the trimmer caps and the Pico (either via headers or directly to the board). The design is published [here](https://oshwlab.com/fruitloop57/fscope-250k5-v2_copy) at OSHWLAB. To view the schematic tap/click one of the 'Open In Editor' buttons (which will open it in EasyEDA).
     
If you're interested in purchasing some boards please email me at scoppy@fhdm.xyz. The price for 5 boards would be AUD$89 including shipping to Canada.
     
Alternatively you could get the boards made yourself. It's just a matter of selecting 'One click order PCB/SMT' from the Fabrication menu in EasyEDA. This will take you to JLCPCB and then you follow the prompts. I can write some instructions if you're interested in going down this route.

.... and I look forward to seeing your design :smile: 

Cheers,
Brett

**Reply by Jason-BCIT** [reply]
Hi Brett,

Yes! I was just got back from holidays and was catching up on your project and saw that a file had been updated "22 days ago"... and that it was very much on point with the circuit that I've been working on. My apologies if my discussion post was a bit premature! Please take it as a sign that your work has me both impressed by and excited about what you are doing!

I'm happy to wait for the online store to be up and running, and would be very interested in ordering five of the boards when you have them in stock. I've spent a fair bit of time researching low-cost 'scope options for my students to build so that they can troubleshoot circuits when studying from home, and as awesome as the DPScope SE has been (and it will remain the option for students with iOS devices) I really like what you have done with Scoppy.

I'm also happy to know that your board has all the options. That makes it a bit easier for me to simplify my design. I'll probably only put the DC/AC input switch and multiple voltage ranges on one channel now. With luck that might let me get it down to a singled sided board design so that we can etch them in the shop. I've got a fair bit of work to do on the final circuit design, but have the basics running on a breadboard right now. There isn't much to look at, but what I've got so far is at https://a360.co/3P6kzYp 

Thank you again for your work, and for sharing it so generously. It has really inspired me to dig into the Pico, and to finally sign up for GitHub. And, believe it or not... OSHWLab is 'something new' for me that I'm very much enjoying checking out!

I'm looking forward to sharing your work with my students in September, and having some of them share it with their future high school students in the years to come.

Thank you again!

Jason

**Reply by fhdm-dev** [reply]
Hi Jason,

The boards have arrived and are now available for purchase at [https://store.fhdm.xyz/fscope-250k5](https://store.fhdm.xyz/fscope-250k5)

**Reply by Jason-BCIT** [reply]
Just put in an order! Looking forward to giving them a shot! Thanks!

**Reply by gilesp1729** [reply]
Do you have boards? The website is still showing out of stock.

**Reply by fhdm-dev** [reply]
Not yet. I'm expecting them to arrive within the next few days.

**Comment by fhdm-dev** [comment]
Cool. I'll put in an order to get some boards manufactured. They should arrive here in about 3 weeks time and I'll make sure the store up and running well before then.

And thanks for sharing your design - nice.
Also, thanks for your interest and enthusiasm in my project. I really do appreciate it :grinning: 

> PS. If anyone else out there is interested in purchasing these boards then please let me know ASAP because I'm only getting a small number manufactured at a time and the lead time is quite long.

**Reply by Jason-BCIT** [reply]
Thanks! I have noticed a few pin reassignments from "stock" scoppy in the fscope's .uf2 file and on the board... the voltage range pins are now GP18-21 and Channel 2 connects to GP28/ADC2 rather than than GP27/ADC1

I'll include those in my design for maximum compatibility.

Jason

**Reply by fhdm-dev** [reply]
... and the signal generator has moved to GPIO 16

**Comment by Jason-BCIT** [comment]
The first round of single-sided board design has worked out well... it is interesting being able to swap out the OpAmps and seeing what they can do. I was trying to use up a bunch of LM324's on this project and was able to get decent enough 0-3V3 performance on the outputs, but then was reminded that the inputs aren't so forgiving of low voltage applications. That and the slew rate was just a bit poky... the TLC274 and MCP6024 both worked better.

By the time I added the logic analyzer inputs (just resistor and Zener for protection) and shrunk the board down a bit I gave up on trying to do a single-sided routing job. My double sided boards should be hear to try out next week.

I've decided to keep it down to one opamp, so am using two channels to buffer inputs one and two, one channel to amplify channel one (for 10x gain or so... my design specs are +/-33V and +/-3.3V operation, but of course that is highly tweakable with different resistor selections. I'm only running Channel 2 at the higher voltage input... although again, that range can be set as desired... maybe +/- 15V would make more sense on Channel 2. The fourth opamp channel is used for amplifying the Signal Generator output, so that it can drive a small speaker, or something, at 5V. The CD4052B allows me to switch the signal generator from pulse to sine... voltage modes 0 and 1 will change the gain on channel one, but give pulsed output, voltage modes 2 and 3 will give the same inputs, but sine output.

At least that's my theory... I'll be testing some of that out next week.

While comparing the board to the older DPScope design, I noticed that Wolfgang had used some neat measurement techniques to get around working with an older, slower, uC. The PIC on his board had a maximum sample rate of 20 ksps, but he re-measured repeating signals multiple times and was able to generate an effective 1 Msps on repeating signals. A neat trick! I think he also used a bit of averaging to knock the noise down a bit. Fusion 360 schematics and pcb still at https://a360.co/3zia7be 

Thanks for making such an interesting project... it has been fun working on this! (Fingers crossed that the two sided boards turn out okay!)

Jason

![20220719_142749](https://user-images.githubusercontent.com/108756994/179855420-3c809c3c-da03-43f4-9f93-781137d588f4.jpg)

**Reply by fhdm-dev** [reply]
That looks awesome Jason. It's exactly the sort of thing I hoped people would use the project for :smile: 

A few people have suggested to me that sampling trick to boost the effective sample rate. I'm still thinking about whether or not to try and implement it.

And FYI, the next update of the firmware will probably be using GPIO 14 and 15 for driving a WiFi status LED and a trigger LED. Just thought I'd let you know in case that affects your next board design.

Looking forward to seeing your two-sided boards.....

**Reply by Jason-BCIT** [reply]
Thanks for the update on the pins! I was telling my wife, "I always feel a bit lazy 'suggesting ideas' about Open Source projects... the whole point of Open Source is that if I've got an idea on how to improve something that I should pitch in and help implement it." But this may be a bit beyond my coding skill, I'm afraid.

One neat idea for the board, that I picked up from the DP Scope project, was for the terminal connectors. Wolfgang used both a BNC connector and an RCA connector and placed the footprints over top of each other... as you can see in the photo, the footprints are surprisingly compatible! This way whoever assembles the board can choose either a BNC connector or an RCA connector. We tend to use the RCA connectors and just build our own 1x probes as that is a bit of a lower cost option and doesn't really impact the overall performance of the board that much. He also added header sockets so that it was easy to use a jumper wire to connect the scope to a breadboard. In one of his later designs he included the option for a header strip so that the scope board could be plugged directly into a breadboard. https://www.picaxestore.com/osc001  

Hmm... there I go making 'suggestions' rather than 'contributions' again! I'm guess I'm just excited about what you've got happening here! :-)

**Reply by fhdm-dev** [reply]
No problem about not contributing code. I prefer to write all of the code myself anyway :)
Suggestions are more than welcome though!

Regarding the RCA connection, I can't quite see what the footprint looks like. Do you have specifications for the particular RCA jack you are using? If I can find something similar in EasyEDA I'll see if it can be added to the next revision of the board.

And you should be able to connect the FScope board to a breadboard with jumpber wires. There's holes for header pins (or sockets) on the inputs and also along both sides of the Pico. If I've missed something there please let me know.

**Reply by Jason-BCIT** [reply]
I totally understand about the code... I just know the feeling of having built something and then immediately having others start offering suggestions on how to 'make it better'... I often get the feeling of "Geez, folks... can we just take a moment to appreciate it BEFORE you start planning my next upgrade??"  :-) 

The RCA jack that I use is from DigiKey https://www.digikey.ca/en/products/detail/cui-devices/RCJ-012/408488?s=N4IgTCBcDaIMIAUC0BGALABhUgcgERAF0BfIA   I suspect there are a few similar ones out there.

My two-sided boards arrived today, so I'm off to the shop to give them a shot! I'll keep you posted!
